### PR TITLE
fix some bugs for v3

### PR DESCRIPTION
**Pull request by mohlek**
Fixed a few bugs as marked in sonar

### CODE REVIEW

File: katharsis-core/src/main/java/io/katharsis/core/internal/resource/AnnotatedFieldBuilder.java [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-import java.lang.reflect.Field;
-import java.lang.reflect.InvocationTargetException;
-
 import com.fasterxml.jackson.databind.introspect.AnnotatedClass;
 import com.fasterxml.jackson.databind.introspect.AnnotatedField;
 import com.fasterxml.jackson.databind.introspect.AnnotationMap;
-
 import io.katharsis.errorhandling.exception.InternalException;
 
+import java.lang.reflect.Constructor;
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/core/internal/resource/AnnotatedFieldBuilder.java]: Any purpose for these changes?

Comment (mohlek) [inline katharsis-core/src/main/java/io/katharsis/core/internal/resource/AnnotatedFieldBuilder.java]: Oh sorry. I didn't mean to push it. Sonar stated a Bug in this file too but i couldn't solve it that quickly. I guess IntelliJ auto formatted my code :-/

File: katharsis-core/src/main/java/io/katharsis/core/internal/repository/adapter/ResourceRepositoryAdapter.java [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
 import io.katharsis.repository.ResourceRepositoryV2;
 import io.katharsis.repository.filter.RepositoryFilterContext;
-import io.katharsis.repository.ResourceRepositoryV2;
 import io.katharsis.repository.request.HttpMethod;
 import io.katharsis.repository.request.QueryAdapter;
 import io.katharsis.repository.request.RepositoryRequestSpec;
 import io.katharsis.repository.response.JsonApiResponse;
 import io.katharsis.resource.information.ResourceInformation;
 
+import java.io.Serializable;
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/core/internal/repository/adapter/ResourceRepositoryAdapter.java]: Why is this here?

File: katharsis-core/src/main/java/io/katharsis/core/internal/resource/IncludeLookupUtil.java [hunk 3]
```diff
@@ -141,7 +135,7 @@ public boolean isInclusionRequested(QueryAdapter queryAdapter, List<ResourceFiel
 					return true;
 				}
 				// FIXME subtyping 
-				if (querySpec != null && contains(querySpec, path)) {
+				if (contains(querySpec, path)) {
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/core/internal/resource/IncludeLookupUtil.java]: is this the fix?

**Comment by chb0github** [comment]
v3 code is in master. And, as discussed before, the `develop` branch is no longer to be used.

File: katharsis-core/src/main/java/io/katharsis/core/internal/resource/IncludeLookupUtil.java (hunk 3 above)
Comment (mohlek) [inline katharsis-core/src/main/java/io/katharsis/core/internal/resource/IncludeLookupUtil.java]: Well this is no "fix". Sonar marked this as `querySpec` is always != null

**Comment by mohlek** [comment]
I don't have an issue for this if you mean that by bug number. I just setup my sonar and "fixed" a few bugs which sonar suggest.
This was once made up in a team meeting. I had barely until now and just wanted to do a little janitoring

**Comment by chb0github** [comment]
Ok, cool. Let me know when you want to merge

**Comment by Ramblurr** [comment]
Is there a public sonar instance we could look at?

**Comment by mohlek** [comment]
Apparently not. I ran this on my company's sonar server. I'll setup a sonar server on one of my vps in the next few days. I'll link it here

**Comment by mohlek** [comment]
I can take a look at it.  
For my projects at work I'm using GitLab with GitLab CI. My script looks like this (just a note for later):

```yml
Sonar preview:
  stage: sonar
  script:
    - mvn --batch-mode sonar:sonar -Dsonar.host.url=$SONAR_URL -Dsonar.analysis.mode=preview -Dsonar.issuesReport.console.enable=false -Dsonar.gitlab.commit_sha=$CI_BUILD_REF -Dsonar.gitlab.ref_name=$CI_BUILD_REF_NAME
  except:
    - master
    - tags
    - /^release-.*$/
Sonar:
  stage: sonar
  script:
    - mvn --batch-mode sonar:sonar -Dsonar.host.url=$SONAR_URL -Dsonar.analysis.mode=publish -Dsonar.issuesReport.console.enable=false -Dsonar.gitlab.commit_sha=$CI_BUILD_REF -Dsonar.gitlab.ref_name=$CI_BUILD_REF_NAME
  only:
    - master
```

Im not familiar with travis so it might take a bit longer. But the getting started page at sonarqube looks very promising.
Also I need to know if there is a auth token already available

**Comment by chb0github** [comment]
I think the build issue is related to the new UI code and npm

**Comment by chb0github** [comment]
Build issue is fixed. What's status on this?

**Comment by chb0github** [comment]
Hello?

**Comment by mohlek** [comment]
Sorry, I had some trouble with sonar authorization (still not fixed yet).
I'll close this issue and start a clean issue with just the sonar integration so we can discuss my problem.

**Comment by mohlek** [comment]
ref #389
